tools/Findpicotool.cmake

```
@@ -34,10 +34,16 @@ if (NOT TARGET picotool)
             )
         endif()
 
+        if (NOT PICOTOOL_GIT_REPOSITORY_URL)
+            set(PICOTOOL_GIT_REPOSITORY_URL https://github.com/raspberrypi/picotool.git)
+        endif()
+        if (NOT PICOTOOL_GIT_BRANCH)
+            set(PICOTOOL_GIT_BRANCH develop)
```

Hmmm, this suddenly reminded me of #2401
@will-v-pi Is the strategy that pico-sdk `develop` always uses picotool `develop`, and that pico-sdk `master` always points at a specific tag of picotool? Is there any way of ensuing that that always remains the case when pico-sdk `master` gets updated from pico-sdk `develop` ?

---

Confirmed that this works on both the current master and develop branches, and on 2.1.0 release, so should be good to add
```suggestion
    if (PICO_SDK_VERSION_PRE_RELEASE_ID)
        set(PICOTOOL_GIT_BRANCH ${PICO_SDK_VERSION_PRE_RELEASE_ID})
    else()
        set(PICOTOOL_GIT_BRANCH ${PICO_SDK_VERSION_STRING})
    endif()
```

---

> I guess there's no guarantee that https://github.com/raspberrypi/picotool.git will always have a branch corresponding to whatever PICO_SDK_VERSION_PRE_RELEASE_ID might get set to? But of course that's irrelevant if PICO_SDK_VERSION_PRE_RELEASE_ID is guaranteed to always be either unset or set to develop.

Or if you set it to something else (for some reason) you probably need to make a corresponding `picotool` branch anyway

> Are they guaranteed to always be incremented in lockstep? If there's been changes to pico-sdk but no changes to picotool, then e.g. pico-sdk 2.1.3 might need picotool 2.1.2 ?

Plan is to always increment them in lockstep - but if that changes then the code here can just be changed when that happens